ctapipe/irf/optimise.py

```diff
@@ -0,0 +1,149 @@
+import operator
+
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable, Table
+from pyirf.binning import create_bins_per_decade
+from pyirf.cut_optimization import optimize_gh_cut
+from pyirf.cuts import calculate_percentile_cut, evaluate_binned_cut
+
+from ..core import Component, QualityQuery
+from ..core.traits import Float
+
+
+class OptimisationResult:
+    def __init__(self, gh_cuts=None, offset_lim=None):
+        self.gh_cuts = gh_cuts
+        if gh_cuts:
+            self.gh_cuts.meta["extname"] = "GH_CUTS"
+        if offset_lim and isinstance(offset_lim[0], list):
+            self.offset_lim = offset_lim
+        else:
+            self.offset_lim = [offset_lim]
+
+    def write(self, out_name, precuts, overwrite=False):
+        if isinstance(precuts, QualityQuery):
```

Comment:
I don't understand the branching here. `precuts` is required, but you check its type and if it is anything but a `QualityQuery` you ignore whatever the user passed in and create a new empty quality query...

Reply:
Yeah, this part is very rushed so I suspect something simply got lost.